### ISSUE TITLE
[BUG] Getting Videos Urls from a Channel have an Index Error

### ISSUE THREAD

**Issue by KomisD** [pytube/pytube#1885]
**Before creating an issue**

Please confirm that you are on the latest version of pytube by installing from the source.
You can do this by running `python -m pip install git+https://github.com/pytube/pytube`.
Sometimes, the pypi library repository is not up to date, and your issue may have been fixed already!

**Describe the bug**
Creating a Youtube channel and then running the command c.video_urls causing and Index error

**To Reproduce**
Here Is my code, well aligned with the documendation:

```
channel = Channel('https://www.youtube.com/c/ProgrammingKnowledge')
for url in c.video_urls[:3]:
   print(url)
```

**Expected behavior**
The expected would be to get the 3 latest video's urls from the channel like this:

['https://www.youtube.com/watch?v=tMqMU1U2MCU',
'https://www.youtube.com/watch?v=YBfInrtWq8Y',
'https://www.youtube.com/watch?v=EP9WrMw6Gzg']

**Output**
---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/pytube/helpers.py](https://localhost:8080/#) in __getitem__(self, key)
     56             try:
---> 57                 next_item = next(self.gen)
     58             except StopIteration:

StopIteration: 

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
1 frames
[/usr/local/lib/python3.10/dist-packages/pytube/helpers.py](https://localhost:8080/#) in __getitem__(self, key)
     58             except StopIteration:
     59                 # If we can't find enough elements for the slice, raise an IndexError
---> 60                 raise IndexError
     61             else:
     62                 self._elements.append(next_item)

IndexError:

**System information**
Please provide the following information:
- Python version (run `python --version`)
Python 3.10.11
- Pytube version (run `print(pytube.__version__)` in python)
15.0.0
- Command used to install pytube
!pip install pytube

**Comment by Omay238** [comment]
Yeah, it isn't just an issue with the index, I believe some YouTube API changed, so the channel.videos on pytube doesn't work currently.

**Comment by benlloydg** [comment]
any update on if channels will be supported again in the future ?
